pkg/scheduler/tests/scheduler_recovery_test.go

```diff
@@ -0,0 +1,398 @@
+/*
+Copyright 2019 Cloudera, Inc.  All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tests
+
+import (
+	cacheInfo "github.com/cloudera/yunikorn-core/pkg/cache"
+	"github.com/cloudera/yunikorn-core/pkg/common/configs"
+	"github.com/cloudera/yunikorn-core/pkg/common/resources"
+	"github.com/cloudera/yunikorn-core/pkg/entrypoint"
+	"github.com/cloudera/yunikorn-scheduler-interface/lib/go/si"
+	"gotest.tools/assert"
+	"testing"
+)
+
+// Test basic interactions from rm proxy to cache and to scheduler.
+func TestSchedulerRecovery(t *testing.T) {
```

Comment:
This a good case test. Would be good to test the bad scenarios too.
what happens if a partition, queue or app does not exist ?

Reply:
This is a good suggestion, let me see how to add some negative tests. Thank you.

Reply:
newly added tests cover this